### PR TITLE
Add Cisco Spark provider

### PR DESCRIPTION
**Pull request by robert-shade**
OAuth Provider for https://www.ciscospark.com/

### CODE REVIEW

File: src/AspNet.Security.OAuth.CiscoSpark/CiscoSparkAuthenticationHelper.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        /// <summary>
+        /// Gets the login corresponding to the authenticated user.
+        /// </summary>
+        public static string GetUserName([NotNull] JObject user) => user.Value<string>("displayName");
+
+        /// <summary>
+        /// Gets the email corresponding to the authenticated user.
+        /// </summary>
+        public static string GetEmail([NotNull] JObject user)  => (string)((JArray)user["emails"]).First;
+
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.CiscoSpark/CiscoSparkAuthenticationHelper.cs]: Please remove the empty line.

File: src/AspNet.Security.OAuth.CiscoSpark/CiscoSparkAuthenticationHelper.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        /// <summary>
+        /// Gets the login corresponding to the authenticated user.
+        /// </summary>
+        public static string GetUserName([NotNull] JObject user) => user.Value<string>("displayName");
+
+        /// <summary>
+        /// Gets the email corresponding to the authenticated user.
+        /// </summary>
+        public static string GetEmail([NotNull] JObject user)  => (string)((JArray)user["emails"]).First;
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.CiscoSpark/CiscoSparkAuthenticationHelper.cs]: This looks a bit fragile. Any chance you could opt for this kind of approach?
https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/src/AspNet.Security.OAuth.Spotify/SpotifyAuthenticationHelper.cs#L35

**Comment by PinpointTownes** [comment]
Thanks for your contribution :smile: 
Except my remark about the `GetEmail` helper, it looks fine.

Note that you'll have to rebase your PR as the .sln file was modified when merging other providers.

**Comment by robert-shade** [comment]
Updated as requested

**Comment by PinpointTownes** [comment]
Thanks, merged: https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/commit/21fb5ff2ee26c4f8d1cdee06c3dc0784c9a095a6 :+1: 

(sorry for the delay... :sweat:)

**Pull request closed without merging**